pkg/cache/memcached.go

```diff
@@ -90,12 +90,24 @@ func memcacheStatusCode(err error) string {
 
 // Fetch gets keys from the cache. The keys that are found must be in the order of the keys requested.
 func (c *Memcached) Fetch(ctx context.Context, keys []string) (found []string, bufs [][]byte, missed []string) {
+	select {
```

Comment:
Can the same error still occur in the underlying `fetch` call, it looks like around line 113 it is throwing away the context.

Reply:
fetch() is only called in one place which is here. i think i'll just collapse these into one function.

i was tempted to simply remove `Fetch()` b/c it's not used in our codebase. I started down this path but it broke a lot of tests. I decided to just leave this call alone in order to keep the PR simple.

done!